src/main/java/org/prebid/server/bidder/evolution/EvolutionBidder.java

```diff
@@ -0,0 +1,96 @@
+import org.prebid.server.util.HttpUtil;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Evolution {@link Bidder} implementation.
+ */
+public class EvolutionBidder implements Bidder<BidRequest> {
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public EvolutionBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        return Result.withValue(createRequest(request));
+    }
+
+    private HttpRequest<BidRequest> createRequest(BidRequest request) {
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .payload(request)
+                .body(mapper.encode(request))
+                .build();
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(bidResponse), Collections.emptyList());
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            throw new PreBidException("Empty seatbid");
+        }
+        return bidsFromResponse(bidResponse);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidResponse bidResponse) {
+        final List<Bid> bidList = bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .findFirst()
+                .map(SeatBid::getBid)
+                .orElse(null);
+        return Stream.of(bidList)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> BidderBid.of(bid, getBidMediaType(bid), bidResponse.getCur()))
+                .collect(Collectors.toList());
+    }
+
+    private BidType getBidMediaType(Bid bid) {
+        try {
+            final EvolutionBidResponse evolutionBidResponse = mapper.mapper().convertValue(
+                    bid.getExt(), EvolutionBidResponse.class);
+            return evolutionBidResponse != null ? evolutionBidResponse.getMediaType() : BidType.banner;
+        } catch (IllegalArgumentException e) {
+            return BidType.banner;
+        }
+    }
```

Comment:
Let's prepare some updates for this methods
```java
 private List<BidderBid> bidsFromResponse(BidResponse bidResponse) {
        final SeatBid firstSeatBid = bidResponse.getSeatbid().get(0);
        return CollectionUtils.emptyIfNull(firstSeatBid.getBid()).stream()
                .filter(Objects::nonNull)
                .map(bid -> BidderBid.of(bid, getBidMediaType(bid.getExt()), bidResponse.getCur()))
                .collect(Collectors.toList());
    }

    private BidType getBidMediaType(JsonNode bidExt) {
        final JsonNode mediaTypeNode = bidExt != null ? bidExt.get("mediaType") : null;
        try {
            return ObjectUtils.defaultIfNull(mapper.mapper()
                    .convertValue(mediaTypeNode, BidType.class), BidType.banner);
        } catch (IllegalArgumentException e) {
            return BidType.banner;
        }
    }
```
Then EvolutionBidResponse class can be deleted